console/client/src/layout/IDELayout.tsx

```diff
@@ -59,52 +53,65 @@ export function IDELayout() {
     })
   }
 
+  // Handle loading a page with the tab query parameters
   React.useEffect(() => {
-    const validateTabs = async () => {
-      const modules = await client.getModules({})
-      const msg = invalidTab({id, type})
-      if (msg) {
-        // IDs an invalid tab ID and type fallback to timeline
-        setActiveTab({id: timelineTab.id, type: timelineTab.type})
-        // On intial mount we have no query params set for tabs so we want to skip setting invalidTabMessage
-        if (type === null && id === null) return
-        return setInvalidTabMessage(msg)
+    const msg = invalidTab({id, type})
+    if (msg) {
+      // Default fallback to timeline
+      setActiveTab({id: timelineTab.id, type: timelineTab.type})
+      if (type === null && id === null) return
+
+      return setInvalidTabMessage(msg)
+    }
+    // Handle timeline tab id
+    if (id === timelineTab.id) {
+      return setActiveTab({id: timelineTab.id, type: timelineTab.type})
+    }
+    if (modules.length) {
+      const ids = id.split('.')
+      // Handle edge case where the id contains and invalid module or verb
+      if (modules.length) {
```

Comment:
There should be a check for `ids.length == 2` somewhere here I think? Otherwise if it's 1 or >2 the array destructuring below fail no?

Reply:
It's in the `invalidTab` util https://github.com/TBD54566975/ftl/blob/main/console/client/src/utils/invalid-tab.utils.ts#L30-L35